plugins/@grouparoo/onesignal/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,101 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import OneSignal from "onesignal-node";
+import { connect } from "../connect";
+
+class InfoError extends Error {
+  errorLevel: string;
+
+  constructor(message) {
+    super(message);
+    this.errorLevel = "info";
+  }
+}
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+
+  const extUserId = newProfileProperties["external_user_id"];
+  if (!extUserId) {
+    throw new Error(
+      `newProfileProperties[external_user_id] is a required mapping`
+    );
+  }
+
+  const payload: any = { tags: {} };
+
+  // set profile properties, including old ones.
+  let newKeys = Object.keys(newProfileProperties);
+  const oldKeys = Object.keys(oldProfileProperties);
+  const allKeys = new Set([...newKeys, ...oldKeys]);
+
+  if (toDelete) {
+    // When deleting, we just clear the old tags. No real deletion occurs.
+    newKeys = [];
+    newGroups = [];
+  }
+
+  for (const key of allKeys) {
+    if (key === "external_user_id") {
+      // not doing this one, it's not a tag
+      continue;
+    }
+
+    const value = newProfileProperties[key];
+    payload.tags[key] = formatVar(value);
+  }
+
+  // Groups are managed as tags
+  // If user is in group "High Value", the attribute "in_high_value" will be set to 1
+  for (const group of newGroups) {
+    const groupAttribute = getGroupAttribute(group);
+    payload.tags[groupAttribute] = 1;
```

Comment:
any sense if others do "true" or "false" or 1/0 or other here?

Reply:
Opted for 1/0 based on the OneSignal docs https://documentation.onesignal.com/docs/add-user-data-tags#recommended-tags-to-send

![image](https://user-images.githubusercontent.com/4368928/110733992-50f6e980-81fd-11eb-9191-4909f71205ca.png)

Reply:
@bleonard following this, should we transform booleans to 1/0 in the formatVar?

Reply:
I think they are just trying to save space, so let's not be too surprising for the user and just be "true"

Reply:
@bleonard Same for the in_group tag or keep 1/0 there?

Reply:
no strong opinion. I guess lets go with "true"

Reply:
Looks like Segment does true/false